File: forge/src/main/java/io/github/xiewuzhiying/vs_addition/forge/compat/computercraft/ForgePeripheralProvider.java
package io.github.xiewuzhiying.vs_addition.forge.compat.computercraft;

import dan200.computercraft.api.peripheral.IPeripheral;
import dan200.computercraft.api.peripheral.IPeripheralProvider;
import io.github.xiewuzhiying.vs_addition.compats.computercraft.PeripheralCommon;
import net.minecraft.core.BlockPos;
import net.minecraft.core.Direction;
import net.minecraft.world.level.Level;
import net.minecraftforge.common.util.LazyOptional;
import org.jetbrains.annotations.NotNull;

public class ForgePeripheralProvider implements IPeripheralProvider {
    @NotNull
    @Override
    public LazyOptional<IPeripheral> getPeripheral(@NotNull Level level, @NotNull BlockPos blockPos, @NotNull Direction direction) {
        IPeripheral peripheral = (new PeripheralCommon()).getPeripheralCommon(level,blockPos);
        if(peripheral==null) peripheral = (new PeripheralForge().getPeripheralForge(level,blockPos,direction));
        if(peripheral==null) return LazyOptional.empty();
        IPeripheral finalPeripheral = peripheral;
        return LazyOptional.of(() -> finalPeripheral);
    }
}

File: forge/src/main/kotlin/io/github/xiewuzhiying/vs_addition/forge/compats/computercraft/ForgePeripheralProvider.kt
package io.github.xiewuzhiying.vs_addition.forge.compats.computercraft

import dan200.computercraft.api.peripheral.IPeripheral
import dan200.computercraft.api.peripheral.IPeripheralProvider
import io.github.xiewuzhiying.vs_addition.compats.computercraft.PeripheralCommon
import net.minecraft.core.BlockPos
import net.minecraft.core.Direction
import net.minecraft.world.level.Level
import net.minecraftforge.common.util.LazyOptional

object ForgePeripheralProvider: IPeripheralProvider {
    override fun getPeripheral(level: Level, blockPos: BlockPos, direction: Direction): LazyOptional<IPeripheral> {
        val peripheral =
            PeripheralCommon.getPeripheralCommon(level, blockPos) ?:
            PeripheralForge.getPeripheralForge(level, blockPos, direction) ?:
            return LazyOptional.empty()
        return LazyOptional.of { peripheral }
    }
}